### DISCUSSION TITLE
Saturn diy

### DISCUSSION THREAD

**Discussion by ledah90** [darthcloud/BlueRetro#153] [Q&A] [answered]
I tested each connection for continuity and everything passes. My issue is when I connect it to my Saturn, it flickers blue and then shuts off instantly. The red light stays on the entire time like it supposed to. As long nothing is connecting to it, the blue light will stay on as well. If anything tries to connect, it immediately turn off and the device disconnects. Any guess as to what could be the cause of this? 
These are the parts used:
Board: ESP32
Level Shifters: SNM74AHCT125N
Saturn controller extension
I have both controller ports connected, using Universal FW and GND on pins 32, 34, and 39.

**Comment by darthcloud** [comment]
Take pictures in which I can clearly see all connect of your build.

**Comment by ledah90** [comment]
![20220126_194826](https://user-images.githubusercontent.com/7066546/151271559-bedddc9b-df45-4995-b27f-a0da7c9fe33d.jpg)
![20220126_194856](https://user-images.githubusercontent.com/7066546/151271563-a8a1a633-a179-4ee2-b4e9-2abd68e729ba.jpg)
![20220126_194903](https://user-images.githubusercontent.com/7066546/151271566-e0de61c1-5d95-48d6-989f-2720b7c838bb.jpg)
![20220126_194910](https://user-images.githubusercontent.com/7066546/151271568-761e5e45-8325-4646-8b1b-828afdcbacc0.jpg)

This is messy, I really don't know how to take a better picture of what I have currently. I wanted a connection without the parallel connector, so all pins are connected directly instead of through parallel. I done this with all of my other consoles and it works fine.

**Reply by darthcloud** [reply]
Get it to work with usb first, no saturn connection, if that fail remove the IC one by one until you find what bug the esp32. If it fail via usb check the log on serial.

After that maybe only try one player first.

**Comment by ledah90** [comment]
I tried USB and it works as it should without any errors. If I connect it to saturn, it disconnects like before. I removed each IC and it appears that IC 3 is the issue. One controller works just fine with the IC 1,2, and 4 connected. I triple checked all my wiring for that IC and it matches. I have also tried 4 other IC chips as well as swapping the ones around on the board. Any other way to test it out?

**Reply by darthcloud** [reply]
You must have a short or resistive cold joint on that chip.

I would completely unwired it and wire again. Before rewiring make sure resistance between socket pins are all 0.

Also note input level shifter power is 3.3v, outputs use 5v make sure they are on right supply.

**Comment by pmgducati** [comment]
@ledah90  Im working on a saturn adapter also and will be designing a PCB for it.  I am happy to share the gerber files with you or I could ship you PCB as I have to order 5 and only need 1.

**Reply by ledah90** [reply]
Sweet. Sorry I haven't replied in awhile I got sick. That would be awesome if you could ship one to me. How much?

**Reply by pmgducati** [reply]
Dont even worry about it, I will be less than $1 to ship the bare PCB's.  Email me your address pmgducati at gmail.  I should have it ordered next week and it takes ~2 weeks to get the PCBs so I would send them your way Mid March.

**Comment by pmgducati** [comment, marked as answer]
@darthcloud I'm working on the Saturn PCB and noticed that U1D pin 11 is called out as IO0.  This is not broken out on the DB25.  However pin 19 on the DB25 is IO16 and is not referenced on U1A,U2A,U3A or U4A.  I am assuming that these pins need to be the same, but my question is should it be IO0 or IO16?


![Screenshot 2022-02-17 235350](https://user-images.githubusercontent.com/20894227/154626100-26ef806c-96f0-452a-8b18-64d5811e5d56.png)

**Reply by ledah90** [reply]
I had mine connected to IO0 and wasn't working. I just connected it to IO16 after you asked that question and the board works perfectly fine now. So I believe U1D pin 11 should be connected to IO16.

**Reply by pmgducati** [reply]
Thanks for doing some digging!  @darthcloud if you could give the blessing on that, it would be great!

**Reply by darthcloud** [reply]
Yes the TR2 gate need to be on IO16, sorry again for that.

But it's good to find those schematic errors! Will update schematic later today.

**Reply by darthcloud** [reply]
Turn out the Kicad was already ok, it's just I forgot to regenerate the PDF.
Did the update just now :)

**Comment by pmgducati** [comment]
@ledah90 Just wanted to let you know that I just ordered the PCBs!

**Comment by pmgducati** [comment]
@ledah90  I got the PCBs yesterday and just finished testing them with the 1.4FW and everything works great!
![PXL_20220313_203057632](https://user-images.githubusercontent.com/20894227/158078082-1fd1309e-ed90-4966-8519-1fce93834e90.jpg)
![PXL_20220313_203102786](https://user-images.githubusercontent.com/20894227/158078083-8c6807bd-fd8e-4538-9cb5-4dd381487226.jpg)
![PXL_20220313_203108489](https://user-images.githubusercontent.com/20894227/158078085-38d32a93-f73e-4722-9fcb-9dabec5eb602.jpg)

**Reply by ledah90** [reply]
Awesome looks great

**Reply by pmgducati** [reply]
One note, U2A is reversed from the others. (you can see it in the last pic.) make sure the orientation is correct before soldering!

**Reply by darthcloud** [reply]
Awesome well done!!
